src/DotVVM.Framework/Resources/Scripts/DotVVM.ts

```diff
@@ -755,6 +755,28 @@ class DotVVM {
         });
     }
 
+    public buildUrlSuffix(urlSuffix: string, query: any): string {
+        var resultSuffix, hashSuffix;
+        if (urlSuffix.indexOf("#") !== -1) {
+            resultSuffix = urlSuffix.substring(0, urlSuffix.indexOf("#"));
+            hashSuffix = urlSuffix.substring(urlSuffix.indexOf("#"));
+        } else {
+            resultSuffix = urlSuffix;
+            hashSuffix = "";
+        }
+        for (var property in query) {
+            if (query.hasOwnProperty(property)) {
+                if (!property) continue;
+                var queryParamValue = ko.unwrap(query[property]);
+                if (!queryParamValue) continue;
```

Comment:
This will break when the query would contain number `0` or boolean `false`, you should probably use `queryParamValue != null` condition. It will `continue;` for `null` and `undefined` and will include the parameter for `0`, empty string and `false`